Makefile.in

```diff
@@ -58,16 +59,64 @@ MANPAGES=dmtcp.1.gz dmtcp_coordinator.1.gz dmtcp_command.1.gz dmtcp_launch.1.gz
 # Macros TEST and XTERM_E used on command line by check1, check2, ...:
 #   make TEST=readline XTERM_E="xterm -e" check-readline
 
+ifeq ($(MULTILIB),1)
+default: display-build-env add-git-hooks build-multilib
+else
 default: display-build-env add-git-hooks build
+endif
+
+build-multilib: build-multilib-m32 build-multilib-m64
+
+build-multilib-m32: config.status-multilib-m32
+	./config.status-multilib-m32
+	${MAKE} -f Makefile clean # Makefile changed; use '-f'
+	${MAKE} -f Makefile build # Makefile changed; use '-f'
+	# ${MAKE} -f Makefile install # Makefile changed; use '-f'
+	@ echo "******************************************************"
+	@ echo "*" && echo "*" && echo "*" && echo "*" && echo "*" && echo "*"
+	@ echo "* FINISHED COMPILING 32-bit MODE. DO 'make -j' FOR 64-bit MODE."
+	@ echo "*" && echo "*" && echo "*" && echo "*" && echo "*" && echo "*"
+	@ echo "******************************************************"
+
+config.status-multilib-m32: config.status
+	cat config.status | \
+	  sed -e s'%S\["M32"\]=.*%S["M32"]="1"%' \
+	      -e s'%D\["CONFIG_MULTILIB"\]="\(.*\)"%D["CONFIG_MULTILIB"]="\1"\nD["CONFIG_M32"]="\1"%' \
+	      -e s'%S\["CONFIG_M32_TRUE"\]=.*%S["CONFIG_M32_TRUE"]=""%' \
+	      -e s'%S\["CONFIG_M32_FALSE"\]=.*%S["CONFIG_M32_FALSE"]="#"%' \
+	      -e s'%S\["CFLAGS"\]="\(.*\)"%S["CFLAGS"]="\1 -m32 -march=i686 -Wa,--32"%' \
+	      -e s'%S\["CXXFLAGS"\]="\(.*\)"%S["CXXFLAGS"]="\1 -m32 -march=i686 -Wa,--32"%' \
+	      -e s'%S\["LDFLAGS"\]="\(.*\)"%S["LDFLAGS"]="\1 -m32 -march=i686 -Wl,-m32 -Wl,-melf_i386 -Wa,--32"%' \
+	      -e s'%D\["ELF_INTERPRETER"\]=.*"%D["ELF_INTERPRETER"]=" \\"/lib/ld-linux.so.2\\""%' \
+	      -e s'%S\["prefix"\]=".*"%S["prefix"]="'$$PWD'/build"%' \
+	  - \
+	> config.status-multilib-m32
+	chmod a+x ./config.status-multilib-m32
+
+# buld-multilib-m{32,64} were delcared .PHONY targets
+# Adding build-multi-m32 as prerequisite allows use of 'make -j'
+build-multilib-m64: config.status-multilib-m64 build-multilib-m32
+	./config.status-multilib-m64 # Restore non-m32 configuration
+	${MAKE} -f Makefile clean # Makefile changed; use '-f'
+	${MAKE} -f Makefile build # Makefile changed; use '-f'
+	# ${MAKE} -f Makefile install # Makefile changed; use '-f'
+
+config.status-multilib-m64: config.status
+	cat config.status | \
+	  sed -e s'%S\["prefix"\]=".*"%S["prefix"]="'$$PWD'/build"%' \
+	  - \
+	> config.status-multilib-m64
+	chmod a+x ./config.status-multilib-m64
 
 mkdirs:
 	$(MKDIR_P) $(targetdir)/bin
 	$(MKDIR_P) $(targetdir)/lib/dmtcp
 
 build: mkdirs dmtcp plugin contrib
 ifeq ($(M32),1)
-	cd lib/dmtcp && $(LN_S) 32/lib/dmtcp/* ./
-	mkdir -p bin && cd bin && $(LN_S) ../lib/dmtcp/32/bin/* ./
+	cd lib/dmtcp && $(LN_S) 32/lib/dmtcp/* ./ || true
+	mkdir -p bin || true
+	cd bin && $(LN_S) ../lib/dmtcp/32/bin/* ./ || true
```

Comment:
Are the `|| true` statements intentional? Shouldn't we fail if we are unable to create the links? Perhaps you can add `-f` to force symlink creation?

Reply:
I believe already that defined as such on the following line:

https://github.com/dmtcp/dmtcp/blob/bac3977fd0a49e4e95681c52bac0a8d01adb5ac4/Makefile.in#L2